one_electron.py

```diff
@@ -0,0 +1,151 @@
+        print("hd_11", hd_11)
+
+        # Applying nuclear potential to spinorb 1 
+        v_psi_1 = orb.apply_potential(-1.0, potential, spinorb1, prec)
+        V1 = spinorb1.dot(v_psi_1)
+
+        hd_V_11 = hd_11 + V1
+
+        eps = hd_V_11.real
+        
+        print('orbital energy', eps - light_speed**2)
+        
+        if(compute_last_energy):
+            break
+
+        mu = orb.calc_dirac_mu(eps, light_speed)
+        tmp = orb.apply_helmholtz(V1, mu, prec)
+        new_orbital = orb.apply_dirac_hamiltonian(tmp, prec, eps, der)
+        new_orbital *= 0.5/light_speed**2
+        print("============= Spinor before Helmholtz =============")
+        print(spinorb1)
+        print("============= New spinor before crop  =============")
+        print(new_orbital)
+        new_orbital.normalize()
+        new_orbital.cropLargeSmall(prec)       
+
+        # Compute orbital error
+        delta_psi = new_orbital - spinorb1
+        deltasq = delta_psi.squaredNorm()
+        error_norm = np.sqrt(deltasq)
+        print('Orbital_Error norm', error_norm)
+        spinorb1 = new_orbital
+        if(error_norm < prec):
+            compute_last_energy = True
+    return spinorb1
+
+
+def gs_1e_D2(spinorb1, potential, mra, prec, der):
+    print('Hartree-Fock 1e D2')
+
+    error_norm = 1.0
+    compute_last_energy = False
+
+    P = vp.PoissonOperator(mra, prec)
+
+    light_speed = spinorb1.light_speed
+    c2 = light_speed**2
+
+    while(error_norm > prec):
+        print("Applying V")
+        Vpsi = orb.apply_potential(-1.0, potential, spinorb1, prec)
+        VVpsi = orb.apply_potential(-0.5/mc2, potential, Vpsi, prec)
+        beta_Vpsi = Vpsi.beta2()
+        apV_psi = Vpsi.alpha_p(prec, der)
+        ap_psi = spinorb1.alpha_p(prec, der)
+        Vap_psi = orb.apply_potential(-1.0, potential, ap_psi, prec)
+        anticom = apV_psi + Vap_psi
+        anticom *= 1.0 / (2.0 * light_speed)
+        RHS = beta_Vpsi + VVpsi + anticom * (0.5/light_speed)
+
+        anticom.cropLargeSmall(prec)
+        beta_Vpsi.cropLargeSmall(prec)
+        VVpsi.cropLargeSmall(prec)
+
+        print("anticom")
+        print(anticom)
+        print("beta_Vpsi")
+        print(beta_Vpsi)
+        print("VV_psi")
+        print(VVpsi)
+        RHS = beta_Vpsi + anticom + VVpsi
+        RHS.cropLargeSmall(prec)
+        print("RHS")
+        print(RHS)
+
+        cke = spinorb1.classicT()
+        cpe = (spinorb1.dot(RHS)).real
+        print("Classic-like energies: ", cke, cpe, cke + cpe)
+        print("Orbital energy: ", c2 * ( -1.0 + np.sqrt(1 + 2 * (cpe + cke) / c2)))
+        mu = orb.calc_non_rel_mu(cke+cpe)
+        print("mu: ", mu)
+        
+
+        new_spinorb1 = orb.apply_helmholtz(RHS, mu, prec)
+        print("normalization")
+        new_spinorb1.normalize()
+        print("crop")
+        new_spinorb1.cropLargeSmall(prec)
+        new_spinorb1.append(new_spinorb1)
+        new_spinorb1.append(new_spinorb1.ktrs(prec))
```

Comment:
There is only one orbital. No need to use ktrs, nor to "append" the original orbital to a list.

Reply:
OK